### PR TITLE
feat: add test for estimate position

### PR DESCRIPTION
**Pull request by Jiajia-Cui**
closes [issue #2076 ](https://github.com/vegaprotocol/system-tests/issues/2076)

### CODE REVIEW

**Review by cdummett — COMMENTED** [review]
Have reviewed the fist test `test_liquidation_and_estimate_position_calculation_AC001` and lgtm 👍  Remaining tests still need a review.

File: tests/integration/test_trading.py [hunk 1]
```diff
@@ -395,3 +400,950 @@ def test_funding_reward_pool(vega_service_with_market: VegaServiceNull):
...
+        MM_WALLET.name,
+        asset="VOTE",
+        amount=1e4,
+    )
+    vega.wait_fn(1)
+    vega.wait_for_total_catchup()
+    
+    configWithSlippage = MarketConfig()
+    configWithSlippage.set("linear_slippage_factor", str(1))  # Set the linear_slippage_factor to 0.5
+    configWithSlippage.set("quadratic_slippage_factor", str(1))  # Set the quadratic_slippage_factor to 0.5
```
Comment (TomMcL) [inline tests/integration/test_trading.py]: These comments don't seem correct ;)

File: tests/integration/test_trading.py [hunk 2]
```diff
@@ -395,3 +400,950 @@ def test_funding_reward_pool(vega_service_with_market: VegaServiceNull):
...
+        trading_key=trader_b.key_name,
+        market_id=market_id,
+        time_in_force="TIME_IN_FORCE_GTC",
+        order_type="TYPE_LIMIT",
+        side="SIDE_SELL",
+        volume=trader_position,
+        price=500,
+    )
+    vega.wait_for_total_catchup()
+    vega.wait_fn(60)
```
Comment (TomMcL) [inline tests/integration/test_trading.py]: Shouldn't be necessary to forward 60 blocks here I don't think

File: tests/integration/test_trading.py [hunk 3]
```diff
@@ -395,3 +400,950 @@ def test_funding_reward_pool(vega_service_with_market: VegaServiceNull):
...
+            # Check Trader A and Trader B positions are still open
+            assert trader_a_position.open_volume == trader_position
+            assert trader_a_position.unrealised_pnl > 0
+            assert trader_b_position.open_volume == -trader_position                    
+            assert trader_b_position.unrealised_pnl < 0
+
+    market_info = vega.market_info(market_id=market_id)
+    market_data = vega.get_latest_market_data(market_id=market_id)
+    print(f"market_data = {market_data}", type(market_data))
+    print(f"market_info = {market_info}", type(market_info))
```
Comment (TomMcL) [inline tests/integration/test_trading.py]: Wouldn't print here it will just fill up the logs

File: tests/integration/test_trading.py [hunk 4]
```diff
@@ -395,3 +400,950 @@ def test_funding_reward_pool(vega_service_with_market: VegaServiceNull):
...
+                asset_id=asset_id,  
+                market_id=market_id,   
+        )
+        open_orders_TRADER_B = vega.list_orders(
+                key_name=trader_b.key_name,
+                wallet_name=trader_b.wallet_name,
+                market_id=market_id,   
+        )
+        
+        print(f"account_TRADER_B at price {price} = {account_TRADER_B}", f"collateral_traderB = {account_TRADER_B.general + account_TRADER_B.bond + account_TRADER_B.margin}")
```
Comment (TomMcL) [inline tests/integration/test_trading.py]: Can we make all the prints debug logs instead?

File: tests/integration/test_trading.py [hunk 5]
```diff
@@ -395,3 +400,950 @@ def test_funding_reward_pool(vega_service_with_market: VegaServiceNull):
...
+    print(f"market_info = {market_info}", type(market_info))
+    _, estimate_liquidation_price = vega.estimate_position(
+        market_id,
+        open_volume=-100,
+        side=["SIDE_SELL"],
+        price=[market_data.mark_price],
+        remaining=[12],
+        is_market_order=[False],
+        collateral_available=collateral_available,
+    )
```
Comment (TomMcL) [inline tests/integration/test_trading.py:868]: It doesn't look like we ever check here the estimated liquidation price with the final open volume has converged to the estimate with the initial open volume + orders do we?

File: tests/integration/test_trading.py [hunk 6]
```diff
@@ -395,3 +400,950 @@ def test_funding_reward_pool(vega_service_with_market: VegaServiceNull):
...
+    _, estimate_liquidation_price = vega.estimate_position(
+        market_id,
+        open_volume=open_volume_A,
+        side=["SIDE_SELL"],
+        price=[1001,1002],
+        remaining=[1,2],
+        is_market_order=[False, False],
+        collateral_available=collateral_available,
+    )
+    # AC 0012-NP-LIPE-004: An estimate is obtained for a long position with multiple limit sell order with the absolute value of the total remaining size of the orders less than the open volume. The estimated liquidation price with sell orders is higher than that for the open volume only.
```
Comment (TomMcL) [inline tests/integration/test_trading.py]:
```suggestion
    # AC 0012-NP-LIPE-004: An estimate is obtained for a long position with multiple limit sell order with the absolute value of the total remaining size of the orders less than the open volume. The estimated liquidation price with sell orders is lower than that for the open volume only.
```

File: tests/integration/test_trading.py [hunk 7]
```diff
@@ -395,3 +400,950 @@ def test_funding_reward_pool(vega_service_with_market: VegaServiceNull):
...
+    _, estimate_liquidation_price_2_order_filled = vega.estimate_position(
+        market_id,
+        open_volume=open_volume_A,
+        side=["SIDE_SELL"],
+        price=[1002],
+        remaining=[0],
+        is_market_order=[False],
+        collateral_available=collateral_available,
+    )
+    # AC 0012-NP-LIPE-004: As the limit orders get filled the estimated liquidation price for the (updated) open volume converges to the estimate originally obtained with open sell orders
```
Comment (TomMcL) [inline tests/integration/test_trading.py]: Shouldn't this be comparing to `estimate_liquidation_price` ?

File: tests/integration/test_trading.py [hunk 8]
```diff
@@ -395,3 +400,950 @@ def test_funding_reward_pool(vega_service_with_market: VegaServiceNull):
...
+    vega.submit_order(
+        trading_key=PARTY_B.name,
+        market_id=market_id,
+        order_type="TYPE_LIMIT",
+        time_in_force="TIME_IN_FORCE_GTC",
+        side="SIDE_SELL",
+        price=970,
+        volume=initial_volume,
+    )
+    vega.submit_order(
```
Comment (TomMcL) [inline tests/integration/test_trading.py:1346]: Assuming the liquidation occurs at 967 could we add something checking that it doesn't occur above the estimated range first?

**Review by TomMcL — APPROVED** [review]
lgtm

**Pull request merged**